conda/common/io.py

```diff
@@ -689,6 +693,114 @@ def _ensure_dir(self):
             os.makedirs(dirname(self.record_file))
 
 
+Reporter = Dict[str, Union[bool, int, str]]
+DetailRecord = Dict[str, Union[str, int, bool]]
+
+
+class ReporterHandlerBase(ABC):
+    """
+    Base class for all reporter handlers
+    """
+
+    @abstractmethod
+    def detail_view(self, data: DetailRecord, **kwargs) -> str: ...
+
+    @abstractmethod
+    def string_view(self, data: str, **kwargs) -> str: ...
+
+
+class ConsoleHandler(ReporterHandlerBase):
```

Comment:
A comment here would be good to explain how this particular reporter handler behaves.